src/core/db/repository/task.py

```diff
@@ -13,49 +15,52 @@ class TaskRepository(ContentRepository):
         """Получить список задач из категорий на которые подписан пользователь."""
-        tasks = await self._session.execute(
+        tasks = await self._session.scalars(
             select(Task)
             .join(Category)
             .where(Category.users.any(id=user_id))
             .where(Task.is_archived == false())
             .limit(limit)
             .offset(offset)
         )
-        return tasks.scalars().all()
+        return tasks.all()
 
-    async def get_all_user_tasks(self) -> list[Task]:
+    async def get_all_user_tasks(self) -> Sequence[Task]:
         """Получить список задач из категорий на которые подписан пользователь."""
-        return await self._session.scalars(select(Task).options(joinedload(Task.category)))
+        all_tasks = await self._session.scalars(select(Task).options(joinedload(Task.category)))
+        return all_tasks.all()
 
-    async def get_tasks_limit_for_user(self, limit: int, offset: int, user: User) -> list[Task]:
+    async def get_tasks_limit_for_user(self, limit: int, offset: int, user: User) -> Sequence[Task]:
         """Получить limit-выборку из списка всех задач пользователя."""
-        return await self._session.scalars(
+        task_limit_for_user = await self._session.scalars((
             select(Task)
             .join(Category)
             .options(joinedload(Task.category))
             .where(Category.users.any(id=user.id))
             .where(Task.is_archived == false())
             .limit(limit)
-            .offset(offset)
-        )
+            .offset(offset)))
+
+        return task_limit_for_user.all()
 
     async def get_user_tasks_count(self, user: User) -> int:
         """Получить общее количество задач для пользователя."""
-        return await self._session.scalar(
+        user_tasks_count =  await self._session.scalar(
             select(func.count(Task.id))
             .join(Category)
             .where(Category.users.any(id=user.id))
-            .where(Task.is_archived == false())
-        )
+            .where(Task.is_archived == false()))
+
+        return user_tasks_count
 
-    async def get_user_task_id(self, task_id) -> list[Task]:
+    async def get_user_task_id(self, task_id) -> Sequence[Task]:
         """Получить задачу по id из категорий на которые подписан пользователь."""
         task = await self._session.scalars(select(Task).options(joinedload(Task.category)).where(Task.id == task_id))
         return task.first()
 
-    async def get_user_tasks_ids(self, ids: list[int]) -> list[Task]:
+    async def get_user_tasks_ids(self, ids: list[int]) -> Sequence[Task]:
         """Получить список задач по ids из категорий на которые подписан пользователь."""
-        tasks = await self._session.execute(select(Task).options(joinedload(Task.category)).where(Task.id.in_(ids)))
-        return tasks.scalars().all()
+        tasks = await self._session.scalar(select(Task).options(joinedload(Task.category)).where(Task.id.in_(ids)))
```

Comment:
[`scalar`](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.scalar) и [`scalars`](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.scalars) два разных метода, которые работают абсолютно по разному - один возвращает модель ([тут на примере разницы между `.one()` и `.scalar()`](https://stackoverflow.com/questions/55662957/what-is-the-difference-between-one-and-scalar)), второй `ScalarResult`.
Пройдись еще раз по коду, и замени на нужные методы.

Ну, и для первого метода нельзя применить метод `.all()` - он возврощает модель (в данном случае `Task`) у которой нет метода `.all()`. Тебе нужно тут просто заменить на нужный метод и все заработает
```suggestion
        tasks = await self._session.scalars(select(Task).options(joinedload(Task.category)).where(Task.id.in_(ids)))
```